### ISSUE TITLE
File `biblatex.sty' not found.

### ISSUE THREAD

**Issue by jtojnar** [tectonic-typesetting/tectonic#33]
When trying to build my thesis, I get the following error. `fithesis` class [uses](https://github.com/Witiko/fithesis3/blob/d2d75fc22d3e9785ef6dec7a6e1d4b870fb81452/style/mu/base.dtx#L419) `biblatex` with `biber` backend.

```
error: something bad happened inside TeX; its output follows:

===============================================================================
(thesis.tex
LaTeX2e <2016/03/31>
Babel <3.9r> and hyphenation patterns for 3 language(s) loaded.
(fithesis3.cls
Document Class: fithesis3 2016/05/15 v0.3.38 fithesis3 MU thesis class
(rapport3.cls
Document Class: rapport3 2004/06/07 v2.1a NTG LaTeX document class
(ntg10.clo)) (keyval.sty) (etoolbox.sty) (ltxcmds.sty) (ifxetex.sty)
(ifluatex.sty)) (polyglossia.sty (makecmds.sty) (xkeyval.sty (xkeyval (xkvutils
))) (fontspec.sty (expl3.sty (expl3-code.tex) (l3xdvipdfmx.def)) (xparse.sty)
(fontspec-xetex.sty (fontenc.sty (eu1enc.def) (eu1lmr.fd)) (xunicode.sty
(t3enc.def (eu1lmss.fd)) (graphicx.sty (graphics.sty (trig.sty) (graphics.cfg)
(xetex.def (infwarerr.sty))))) (fontspec.cfg)))) (gloss-english.ldf) (url.sty)

! LaTeX Error: File `biblatex.sty' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: sty)

Enter file name: ===============================================================================

error: terminal input forbidden
```

**Comment by pkgw** [comment]
Ah, it looks like I need to add `collection-bibtexextra` to the default bundle.

Relevant code [is here](https://github.com/tectonic-typesetting/tectonic-staging/blob/master/staging.sh#L146) — if you're up for a PR, that would be excellent. But it's a one-line change so I am happy to do it myself.

I'm not a `biblatex` user. I've been unclear on something — does it require a separate command-line program other than the classic `bibtex` program? If so, it would take some effort to integrate that functionality ... but I get the impression that more and more people are using `biblatex` so it may be time to get on board that train.

**Comment by jtojnar** [comment]
As I understand it `biblatex` can use different backends, `fithesis` uses `biber` backend so I need to run `biber thesis.bcf`. It looks like there is also legacy `bibtex` backend: https://tex.stackexchange.com/questions/13509/biblatex-in-a-nutshell-for-beginners

I need to run so if you could add the collection, that would be nice.

**Comment by pkgw** [comment]
Ah, OK, thanks. You can try downloading `biblatex.sty` separately and placing it in the same directory as your document, but probably a lot of support files will be needed.

I will investigate what it would take to add a `biber` engine to the backend, but I have to admit that this will probably be a low priority unless lots of people express interest. I'll file a separate issue.

**Comment by pkgw** [comment]
I'll upload a new bundle tomorrow that should provide the style files. I thought I'd be able to get it done today, but I ran into a limitation in [zip-rs](https://github.com/mvdnes/zip-rs) that it could only handle Zip files with less than 65536 files in them!

**Comment by pkgw** [comment]
This particular should now be fixed with the update to the `2016.0r4` version of the default bundle (cf issue #37). No recompilation of reinstallation of Tectonic is needed, but you may need to blow away the cache of downloaded files, `~/.cache/Tectonic` on Linux and **I think** something like ~/Library/Caches/Tectonic` on OSX. Please reopen if `biblatex.sty` fails to appear after doing that.

It is looking like full `biblatex` support will be a challenge — `biber` is a large, complex Perl program, so there's no simple path forward for including it in the single Tectonic binary. To be investigated.

**Comment by jtojnar** [comment]
Now it fails on a different issue:

```log
! Package biblatex Error: Option 'urldate=edtf' invalid.

See the biblatex package documentation for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.11963 \blx@processoptions
```
